components/drafts/channel_draft/channel_draft.tsx

```diff
@@ -0,0 +1,128 @@
+    channel: Channel;
+    displayName: string;
+    draftId: string;
+    id: $ID<Channel>;
+    status: UserStatus['status'];
+    type: 'channel' | 'thread';
+    user: UserProfile;
+    value: PostDraft;
+}
+
+function ChannelDraft({
+    channel,
+    displayName,
+    draftId,
+    id,
+    status,
+    type,
+    user,
+    value,
+}: Props) {
+    const dispatch = useDispatch();
+    const history = useHistory();
+
+    const teamId = useSelector(getCurrentTeamId);
+    const channelUrl = useSelector((state) => getChannelURL(state, channel, teamId));
+
+    useEffect(() => {
+        if (!channel?.id) {
+            dispatch(getChannel(id));
+        }
+    }, [channel?.id]);
+
+    const onSubmit = useMemo(() => {
+        return makeOnSubmit(channel.id, '', '');
+    }, [channel.id]);
+
+    const handleSubmit = async () => {
+        await dispatch(onSubmit(value));
+        return {data: true};
+    };
+
+    const handleOnEdit = useCallback(() => {
+        history.push(channelUrl);
+    }, [channelUrl]);
+
+    const handleOnSend = useCallback((id: string) => {
+        handleSubmit().then(() => {
+            handleOnDelete(id);
+            handleOnEdit();
+        });
+    }, []);
+
+    const handleOnDelete = useCallback((id: string) => {
+        localStorage.removeItem(id);
+        dispatch(setGlobalItem(id, {message: '', fileInfos: [], uploadsInProgress: []}));
+    }, []);
+
+    if (!channel) {
+        return null;
+    }
+
+    return (
+        <Panel onClick={handleOnEdit}>
+            {({hover}) => (
+                <>
+                    <Header
+                        hover={hover}
+                        actions={(
+                            <DraftActions
+                                channelName={channel.display_name || displayName}
+                                draftId={draftId}
+                                onDelete={handleOnDelete}
+                                onEdit={handleOnEdit}
+                                onSend={handleOnSend}
+                            />
+                        )}
+                        title={(
+                            <DraftTitle
+                                channel={channel}
+                                type={type}
+                                userId={user.id}
+                            />
+                        )}
+                        timestamp={value.createAt}
+                    />
+                    <Body
+                        channelId={channel.id}
+                        displayName={displayName}
+                        draftId={draftId}
+                        fileInfos={value.fileInfos}
+                        message={value.message}
+                        status={status}
+                        uploadsInProgress={value.uploadsInProgress}
+                        user={user}
+                    />
+                </>
+            )}
+        </Panel>
+    );
+}
+
+export default memo(ChannelDraft);
```

Comment:
well I've used this here and in parent, but not in items children,
my mental model on memo is if I connect the component to redux memo it,
and don't memo the children that inherit props from parent, since those probably would also change when the parent changes.

Reply:
That makes sense to me. I imagine we'll want to look into usage of memo later on to come up with a "when to use memo" doc, but that's not something to worry about right now.